spec.bs

```diff
@@ -360,12 +360,12 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
 
             <dl class="switch">
                 :   If it was fulfilled with value |index|:
-                ::  1. If |index| is greater than |urlList|'s [=list/size=], then [=queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
+                ::  1. If |savedQueryName| is a [=string=] that is not the empty string, then run [=store the index for a saved query=] with |window|, |navigable|, |workletDataOrigin|, |moduleURLRecord|, |operationName|, |savedQueryName|, and |index|.
+                    1. If |index| is greater than |urlList|'s [=list/size=], then [=queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
 
                         Note: The result index is beyond the input urls' size. This violates the selectURL() protocol, and we don't know which url should be selected.
 
                     1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=resolve=] |promise| with |index|.
-                    1. If |savedQueryName| is a [=string=] that is not the empty string, then run [=store the index for a saved query=] with |navigable|, |workletDataOrigin|, |moduleURLRecord|, |operationName|, |savedQueryName|, and |index|.
                     1. Run |privateAggregationCompletionTask|.
 
                 :   If it was rejected:
```

Comment:
What should happen to the waiting callbacks if the first selectURL() is rejected (before validating index against urlList’s [size](https://infra.spec.whatwg.org/#list-size))? I don't see they are being handled. IMO, we can/should also cache this response, and return the default index 0.

For queryData’s [index](https://pr-preview.s3.amazonaws.com/WICG/shared-storage/pull/202.html#saved-query-data-index), you might want to distinguish between "unresolved" and "rejected". Or, perhaps you can reuse "0" for the "rejected" status.

Reply:
Let's reuse "0" for the rejected status.

Added below in lines 372-373.